daemon/cmd/deletion_queue.go

```
@@ -69,10 +66,10 @@ func newDeletionQueue(lc cell.Lifecycle, p promise.Promise[*Daemon]) *deletionQu
 	return dq
 }
 
-func (dq *deletionQueue) lock(ctx context.Context) error {
+func (dq *deletionQueue) lock(ctx context.Context) {
 	if err := os.MkdirAll(defaults.DeleteQueueDir, 0755); err != nil {
 		log.WithError(err).WithField(logfields.Path, defaults.DeleteQueueDir).Error("Failed to ensure CNI deletion queue directory exists")
-		return nil
```

I'm curious about this case, why not return the error created by `os.MkdirAll` ?

---

probably because the whole lock is best-effort (at least it's handled this way in the rest of the `lock` method) :thinking:

---

Thanks, good call. The operations following `(*deletionQueue).lock` in `(*deletionQueue).Start` would anyway fail if `os.MkdirAll` fails and the directory `defaults.DeleteQueueDir` doesn't exist.

I'll change to return the error instead in this case.

---

> probably because the whole lock is best-effort (at least it's handled this way in the rest of the lock method) 🤔

Yeah, all of this is best-effort but I guess we will spam the log less with successive errors in case we return the `os.MkdirAll` error.